src/availability/query_data.rs

```diff
@@ -429,3 +429,84 @@ pub(crate) fn payload_size<Types: NodeType>(payload: &Payload<Types>) -> u64 {
         Err(_) => 0,
     }
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "")]
+pub struct BlockSummaryQueryData<Types: NodeType> {
+    pub(crate) header: Header<Types>,
+    pub(crate) hash: BlockHash<Types>,
+    pub(crate) size: u64,
+    pub(crate) num_transactions: u64,
+    pub(crate) proposer_id: SignatureKey<Types>,
+}
+
+// Add some basic getters to the BlockSummaryQueryData type.
+impl<Types: NodeType> BlockSummaryQueryData<Types> {
+    pub fn header(&self) -> &Header<Types> {
+        &self.header
+    }
+
+    pub fn hash(&self) -> BlockHash<Types> {
+        self.hash
+    }
+
+    pub fn height(&self) -> u64 {
+        self.header.block_number()
+    }
+
+    pub fn size(&self) -> u64 {
+        self.size
+    }
+
+    pub fn num_transactions(&self) -> u64 {
+        self.num_transactions
+    }
+
+    pub fn proposer_id(&self) -> SignatureKey<Types> {
+        self.proposer_id.clone()
+    }
+}
+
+// Get the Proposer from the given header.  At the moment the proposer cannot
+// be determined from the header, however it is intended to in the future.
+// For now we will just generate one with dummy data.
+// TODO update this function to retrieve the proposer from the Header type once
+//      the Header type has been updated to include the proposer.
+pub fn get_proposer<Types: NodeType>(_header: &Header<Types>) -> SignatureKey<Types> {
+    use hotshot::types::SignatureKey;
+    let (key, _) = SignatureKey::generated_from_seed_indexed([0; 32], 0);
+    key
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "")]
+pub struct TransactionSummaryQueryData<Types: NodeType> {
+    pub(crate) hash: TransactionHash<Types>,
+    pub(crate) header: Header<Types>,
+    // We want a way to determine a summary for each rollup entry, without
+    // the data directly, but rather a summary of the data.
+    // For now, we'll roll with the `Payload` itself.
+    pub(crate) transaction: Transaction<Types>,
+}
+
+// Since BlockSummaryQueryData can be derived entirely from BlockQueryData, we
+// implement the From trait to allow for a seamless conversion using rust
+// contentions.
+impl<Types: NodeType> From<BlockQueryData<Types>> for BlockSummaryQueryData<Types>
+where
+    Payload<Types>: QueryablePayload,
+{
+    fn from(value: BlockQueryData<Types>) -> Self {
+        let payload = value.payload();
+        let size = payload_size::<Types>(&payload.clone());
+        let num_transactions = payload.len(value.metadata());
+
+        BlockSummaryQueryData {
+            header: value.header().clone(),
+            hash: value.hash(),
+            size,
+            num_transactions: num_transactions as u64,
+            proposer_id: get_proposer::<Types>(value.header()),
```

Comment:
For consistency, I think we should add `num_transactions` and `proposer_id` to `BlockQueryData` as well. Really, `BlockQueryData` should just be
```rust
struct BlockQueryData<Types> {
  summary: BlockSummaryQueryData<Types>,
  payload: Payload<Types>,
}
```

But to keep the JSON responses flat, it might not be a bad idea to just copy the fields of `BlockSummaryQueryData` into `BlockQueryData`. Anyways, if we do that we can just use `value.num_transactions` and `value.proposer_id` here

Reply:
Making this change requires us to be able to populate these values in `BlockQueryData::new`, and outside.  These values are ultimately determined by the `payload` with the `trait` `QueryablePayload`, which ultimately means polluting these `impl` blocks with more instances of the `Payload<Types> : QueryablePayload` restriction.

My current attempts have me attempting to limit these restrictions to `BlockQueryData::new` and `BlockQueryData::gensis` which seems to work.  However, I also need to add it to the `sql.rs` file for `parse_block`, which ultimately spreads to `parse_payload`.

The addition to `BlockQueryData::new` also affects `src/data_source/update.rs:88` which seems like this is leaking too much.

Reply:
Ah yeah that trait bound will cascade everywhere

Reply:
It is actually fine to put that trait bound everywhere it's needed to make `new` work (e.g. `update.rs` would be fine). If it is too much work we could also save this for a follow-up